public/static/app/common/EvgUtil.js

```diff
@@ -0,0 +1,32 @@
+mciModule.factory('EvgUtil', function() {
+  var DATE_ID_FORMAT = 'YY_MM_DD_HH_mm_ss'
+
+  // Replaces all '-' with '_'
+  function normalize(text) {
+    return text.replace(/-/g, '_')
+  }
+
+  // internal function which generates evergreen id from given tokens
+  function generateId() {
```

Comment:
Could we make this function explicitly take in an array of arguments which it will join? The current implementation is hard to follow